src/util/chunked_content_decoder.js

```diff
@@ -82,6 +89,31 @@ class ChunkedContentDecoder extends stream.Transform {
                 } else {
                     this.state = STATE_READ_CHUNK_HEADER;
                 }
+            } else if (this.state === STATE_WAIT_TRAILER_DATA) {
+                // TODO - handle trailers (HTTP headers)
+                for (; index < buf.length; ++index) {
+                    if (buf[index] === CR_CODE) {
+                        if (buf.length === 2 && buf[index + 1] === NL_CODE) {
+                            this.chunk_trailer_str = '';
+                            this.state = STATE_WAIT_NL_END;
+                            break;
+                        } else {
+                            this.state = STATE_WAIT_NL_TRAILER;
+                            break;
+                        }
```

Comment:
```suggestion
                        this.state = STATE_WAIT_NL_TRAILER;
```

Reply:
instead of checking for this.chunk_trailer_str.endsWith('\n') perhaps we better add a new state STATE_CHECK_TRAILER ... I will edit the code suggestion